server/src/main/scala/passengersplits/polling/Polling.scala

```diff
@@ -108,78 +138,194 @@ object AtmosFilePolling {
-                //              statefulPoller.onNewFileSeen(zipFileName)
+  type UnzipFileContentFunc[ZippedFileName] = (ZippedFileName) => Future[List[UnzippedFileContent]]
 
-              })
+  trait BatchFileState {
+    def latestFile: String
 
-            log.info(s"AdvPaxInfo: tickId: ${tickId} updating latestFile: ${latestFile} to ${zipFileName}")
-            latestFile = zipFileName
+    def onZipFileProcessed(filename: String): Unit
 
-            log.info(s"AdvPaxInfo: tickId: ${tickId} finished processing zip $zipFileName $latestFile")
-          })
+    def onBatchComplete(tickId: TickId): Unit
+  }
+
+  trait LogginBatchFileState extends BatchFileState {
+    var latestFileName: String
+
+    def onZipFileProcessed(filename: String) = {
+      log.info(s"setting latest filename ${filename}")
+      latestFileName = filename
+    }
+
+    def onBatchComplete(tickId: TickId) = log.info(s"tickId: $tickId batch complete")
+
+    def latestFile = latestFileName
+  }
+
+
+  def runSingleBatch(tickId: TickId, zipFilenamesSource: Source[String, NotUsed], unzipFileContent: UnzipFileContentFunc[String],
+                     flightPassengerReporter: ActorRef, batchFileState: BatchFileState,
+                     batchAtMost: FiniteDuration)(implicit actorSystem: ActorSystem, materializer: Materializer): Unit = {
+    log.info(s"tickId: $tickId Starting batch")
+    val futureZipFiles: Future[Seq[String]] = zipFilenamesSource.runWith(Sink.seq)
+    val internalMaterializer = materializer //..ActorMaterializer()
+    futureZipFiles.onComplete {
+      case oc => log.info(s"tickId: $tickId zip file list retrieval completes with $oc")
+    }
+    val futureOfFutureProcessedZipFiles: Future[Seq[Future[String]]] = for (fileNames <- futureZipFiles) yield {
+      processSingleZipFile(tickId, unzipFileContent, flightPassengerReporter, batchFileState, fileNames)
+    }
+
+    case class ZipFailure(filename: String, t: Throwable)
+
+    val futureOfProcessedZipFiles: Future[Seq[Try[String]]] = futureOfFutureProcessedZipFiles.flatMap { (batchSuccess: Seq[Future[String]]) =>
+      val allZipFilesInBatch: Future[Seq[Try[String]]] = waitAll(batchSuccess)
+      allZipFilesInBatch.onComplete {
+        (oc) =>
+          log.info(s"tickId: $tickId batchComplete")
+          batchFileState.onBatchComplete(tickId)
       }
+      allZipFilesInBatch
     }
 
-    source.runForeach { tickId =>
-      runSingleBatch(tickId)
+    val allZips = futureOfProcessedZipFiles.onComplete {
+      case Success(allZipFilesInBatch) =>
+        log.info(s"tickId: $tickId batch success $allZipFilesInBatch")
+      case Failure(t) =>
+        log.error("failure in runSingleBatch", t)
     }
+    log.info(s"tickId: $tickId, awaiting zip file completions")
+    Await.result(futureOfProcessedZipFiles, batchAtMost) // todo don't commit this either
```

Comment:
We need at least one. There's now just a big ol' timeout on the batch. *maybe* we'll want more control. My fear is the first, big batch, is likely slower than subsequent. But not sure I want the additional complexity of managing that. Thoughts?